### ISSUE TITLE
Update set_priors() syntax

### ISSUE THREAD

**Issue by clarabicalho** [macartan/gbiqq#19]
To match get_type() statement-type syntax

**Comment by lilymedina** [comment]
Question on confounds and priors. 

Say yo have a model `X->Y` with confound   `list(X = "(Y == 1)")`

Now, if you want to specify prior `"X==1" = 2` (or `X1 = 2`). Would this be assigned to both Xs (i.e `X` for confounded relationship and `X` for non-confound)? Or should the user provide more information on `X `so that gbiqq can pin down the exact variable?



This feature is slowly coming together. It seems to be working fine for a model without confounds butstill figuring it out for a model with confounds.

**Comment by lilymedina** [comment]
I'm having trouble finding a syntax that works for almost any kind query. 

Below is a list of possible approaches to set priors for models with confound. Let me know if you have a preference for any of them or if you have any other suggestion.



1.   `alphas = (list(X = "(Y == 1) = 2"))` Similar to the syntax for setting confounds -- Query would point at confounding vars `X0.1` and `X1.1`)

 -  Substantive Syntax
-  Requires to develop new functions and quite a bit of string manipulation, which I rather not implement at this point, but can definitely be a future development
- Doesn't allow for precision on specification -- how to set prior of `X0.1` only?


2.   `alphas = c(X0.1 = 2, X1.1 = 2)` Only allowing row names of P.

+ Straightforward implementation 
-  Not such informative syntax.

We could change Ps' row names to make them more informative

3.  `"X==1" = 2 #or X1 = 2`

(assigns 2 to both Xs as described in post above)

- substantive meaning 
-  easy to implement
- It doesn't allow for a precise specification. e.g users wouldn't be able to specify queries of the sort "X when Y =1"



I see how a mix of options 2 and 3 can potentially cover most kinds of queries.

**Comment by macartan** [comment]
> Question on confounds and priors.
> 
> Say yo have a model `X->Y` with confound `list(X = "(Y == 1)")`
> 
> Now, if you want to specify prior `alphas = c("X==1" = 2)` (or `X1 = 2`). Should this be assigned to both Xs (i.e `X` for confounded relationship and `X` for non-confound)? Or should the user provide more information on `X `so that gbiqq can pin down the exact variable?
> 

For both
But you should also be able to specify if you need to  say that it is for one or the other X

**Comment by macartan** [comment]
> 
> 1. `alphas = (list(X = "(Y == 1) = 2"))` Similar to the syntax for setting confounds -- Query would point at confounding vars `X0.1` and `X1.1`)
> 
positions of ='s seems a bit funny: I think I get you though

Imagine a more explicit listing of arguments:

```
alphas <- set_alphas(
    param_set = list(X = "Y[X=1] == 1"),            # optional: Indicate which group via confound syntax
    param = TRUE,                         # If TRUE select all params within this set; otherwise particular params
    alpha =  2                   # Value to assign
)
```
 that would put 2 on both X0.1` and `X1.1`

Then a usage might be:
```
model %>%
  set_alphas(parma_set = "Y",  alpha = 4) %>%
  set_alphas(list(X = "Y[X=1] == 1"), alpha = 2) %>% 
  set_alphas(parma = "Y01",  alpha = 5) %>%
```

showing three ways you might use this

> 
> 1. `alphas = c(X0.1 = 2, X1.1 = 2)` Only allowing row names of P.
> 
> * Straightforward implementation
> 
> * Not such informative syntax.
>

Yes, narrow! But this is a special case of the above with "param = X0.1; alpha_value = 2'
 
> We could change Ps' row names to make them more informative
> 
> 1. `"X==1" = 2 #or X1 = 2`

hard to do with the confounds, but also when they get big. eg hard to name Y0010101101.3
> 
> (assigns 2 to both Xs as described in post above)
> 
> * substantive meaning
> * easy to implement
> * It doesn't allow for a precise specification. e.g users wouldn't be able to specify queries of the sort "X when Y =1"
> 
> I see how a mix of options 2 and 3 can potentially cover most kinds of queries.

**Comment by lilymedina** [comment]
I used the script below to build the package with both versions 3.5 and 3.6. 

```
roxygen2::roxygenise(clean = TRUE)
pkgbuild::compile_dll()
remotes::install_github("stan-dev/rstantools")
devtools::install(local = FALSE)
```

**Comment by macartan** [comment]
This picks out all the types where X has a positive effect on Y when Z = 0. 

This picks out all the types where X has a positive effect on Y when Z = 0. 
```
model <- make_model("X -> Y <- Z")  %>%
	set_priors(alphas = list(Y = c("Y[X=1, Z=0] > Y[X=0, Z=0]" = 3))) 
model	
```

This is not accepted however, for the case where I want X having an effect for any value of Z:

```
model <- make_model("X -> Y <- Z")  %>%
	set_priors(alphas = list(Y = c("Y[X=1, Z=.] > Y[X=0, Z=.]" = 3))) 
model	
```

```
 Error in eval(parse(text = paste0(do, collapse = "")), envir = eval_var) : 
  object '.' not found 
```

`
model <- make_model("X -> Y <- Z")  %>%
	set_priors(alphas = list(Y = c("Y[X=1, Z=0] > Y[X=0, Z=0]" = 3))) 
model	
`

**Comment by lilymedina** [comment]
Try adding parentheses to both sides of the inequality.

model <- make_model("X -> Y <- Z")  %>%
	set_priors(alphas = list(Y = c("(Y[X=1, Z=.]) > (Y[X=0, Z=.])" = 3)))




expand_wildcard needs expressions to be contained within parentheses

**Comment by macartan** [comment]
of course; thanks!
I have made that mistake a bunch of times! an informative error message in
expand_wildcard would help

**Comment by lilymedina** [comment]
solved in PR [#48](https://github.com/macartan/gbiqq/pull/48
